pkg/cmd/cli/cmd/rollout/latest.go

```diff
@@ -0,0 +1,160 @@
+by the trigger.`
+
+	rolloutLatestExample = `  # Start a new deployment
+  %[1]s rollout latest dc/nginx
+`
+)
+
+// RolloutLatestOptions holds all the options for the `rollout latest` command.
+// TODO: Support --dry-run
+type RolloutLatestOptions struct {
+	mapper meta.RESTMapper
+	typer  runtime.ObjectTyper
+	infos  []*resource.Info
+
+	out         io.Writer
+	shortOutput bool
+
+	oc              client.Interface
+	kc              kclient.Interface
+	baseCommandName string
+}
+
+// NewCmdRolloutLatest implements the oc rollout latest subcommand.
+func NewCmdRolloutLatest(fullName string, f *clientcmd.Factory, out io.Writer) *cobra.Command {
+	opts := &RolloutLatestOptions{
+		baseCommandName: fullName,
+	}
+
+	cmd := &cobra.Command{
+		Use:     "latest DEPLOYMENTCONFIG",
+		Short:   "Start a new deployment for a config with the latest state from its triggers",
+		Long:    rolloutLatestLong,
+		Example: fmt.Sprintf(rolloutLatestExample, fullName),
+		Run: func(cmd *cobra.Command, args []string) {
+			err := opts.Complete(f, cmd, args, out)
+			kcmdutil.CheckErr(err)
+
+			if err := opts.Validate(); err != nil {
+				kcmdutil.CheckErr(kcmdutil.UsageError(cmd, err.Error()))
+			}
+
+			err = opts.RunRolloutLatest()
+			kcmdutil.CheckErr(err)
+		},
+	}
+
+	kcmdutil.AddOutputFlagsForMutation(cmd)
+
+	return cmd
+}
+
+func (o *RolloutLatestOptions) Complete(f *clientcmd.Factory, cmd *cobra.Command, args []string, out io.Writer) error {
+	if len(args) != 1 {
+		return errors.New("one deployment config name is needed as argument.")
+	}
+
+	namespace, _, err := f.DefaultNamespace()
+	if err != nil {
+		return err
+	}
+
+	o.oc, o.kc, err = f.Clients()
+	if err != nil {
+		return err
+	}
+
+	o.mapper, o.typer = f.Object(false)
+	o.infos, err = resource.NewBuilder(o.mapper, o.typer, resource.ClientMapperFunc(f.ClientForMapping), f.Decoder(true)).
+		ContinueOnError().
+		NamespaceParam(namespace).
+		ResourceNames("deploymentconfigs", args[0]).
+		SingleResourceType().
+		Do().Infos()
+	if err != nil {
+		return err
+	}
+
+	o.out = out
+	o.shortOutput = kcmdutil.GetFlagString(cmd, "output") == "name"
+
+	return nil
+}
+
+func (o RolloutLatestOptions) Validate() error {
+	if len(o.infos) != 1 {
+		return errors.New("a deployment config is required.")
+	}
+	return nil
+}
+
+func (o RolloutLatestOptions) RunRolloutLatest() error {
+	info := o.infos[0]
+	config, ok := info.Object.(*deployapi.DeploymentConfig)
+	if !ok {
+		return fmt.Errorf("%s is not a deployment config", info.Name)
+	}
+
+	// TODO: Consider allowing one-off deployments for paused configs
+	if config.Spec.Paused {
+		return fmt.Errorf("cannot deploy a paused deployment config")
+	}
+
+	deploymentName := deployutil.LatestDeploymentNameForConfig(config)
+	deployment, err := o.kc.ReplicationControllers(config.Namespace).Get(deploymentName)
+	switch {
+	case err == nil:
+		// Reject attempts to start a concurrent deployment.
+		if !deployutil.IsTerminatedDeployment(deployment) {
```

Comment:
there should be an utility method for this check, like `deployutil.IsConcurrentDeployment()` or something similar.